Microsoft.Toolkit.Uwp.UI.Controls.Primitives/ConstrainedBox/ConstrainedBox.Properties.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Dependency properties for the <see cref="ConstrainedBox"/> class.
+    /// </summary>
+    public partial class ConstrainedBox
+    {
+        /// <summary>
+        /// Gets or sets the scale for the width of the panel. Should be a value between 0-1.0. Default is 1.0.
+        /// </summary>
+        public double ScaleX
+        {
+            get { return (double)GetValue(ScaleXProperty); }
+            set { SetValue(ScaleXProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="ScaleX"/> property.
+        /// </summary>
+        public static readonly DependencyProperty ScaleXProperty =
+            DependencyProperty.Register(nameof(ScaleX), typeof(double), typeof(ConstrainedBox), new PropertyMetadata(1.0, ConstraintPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets the scale for the height of the panel. Should be a value between 0-1.0. Default is 1.0.
+        /// </summary>
+        public double ScaleY
+        {
+            get { return (double)GetValue(ScaleYProperty); }
+            set { SetValue(ScaleYProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="ScaleY"/> property.
+        /// </summary>
+        public static readonly DependencyProperty ScaleYProperty =
+            DependencyProperty.Register(nameof(ScaleY), typeof(double), typeof(ConstrainedBox), new PropertyMetadata(1.0, ConstraintPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets the integer multiple that the width of the panel should be floored to. Default is null (no snap).
+        /// </summary>
+        public int MultipleX
+        {
+            get { return (int)GetValue(MultipleXProperty); }
+            set { SetValue(MultipleXProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="MultipleX"/> property.
+        /// </summary>
+        public static readonly DependencyProperty MultipleXProperty =
+            DependencyProperty.Register(nameof(MultipleX), typeof(int), typeof(ConstrainedBox), new PropertyMetadata(null));
+
+        /// <summary>
+        /// Gets or sets the integer multiple that the height of the panel should be floored to. Default is null (no snap).
+        /// </summary>
+        public int MultipleY
+        {
+            get { return (int)GetValue(MultipleYProperty); }
+            set { SetValue(MultipleYProperty, value); }
+        }
```

Comment:
Does this need to be an integer? What if I want the value to be a multiple of 3.5.

Reply:
I suppose, but why would you want to snap to an un-even number of pixels?